### DISCUSSION TITLE
How to keep pivot data between rehydrations and send with event

### DISCUSSION THREAD

**Discussion by SturmB** [livewire/livewire#5193] [1. Help]
When I check the checkbox of a Livewire Component, it suddenly loses the pivot data:

![image](https://user-images.githubusercontent.com/6216224/185822054-89007f33-9018-4096-9359-876140f0f783.png)

I know this has been addressed at #3487, but the accepted solution is difficult for a newbie like me to understand. Especially since I would like to pass the model and its associated pivot data as part of an event to another Livewire Component:

```php
use App\Models\Resource;
use Livewire\Component;

class ResourceCheckbox extends Component
{
    public bool $checked;
    public Resource $resource;

    public function render()
    {
        return view("livewire.resource-checkbox");
    }

    public function mount()
    {
        $this->checked = false;
    }

    public function checkboxClicked()
    {
        $this->emit(
            "checkboxClicked",
            $this->checked,
            $this->resource->id,
        );
    }
}
```

Is there perhaps some way I can add this data to the model so that it will persist on the other end of the event (the child component with the listener)? Does the "computed property" capability do that? Or does that computed property only exist in the parent component and cannot be sent to the child component via an event?

I'd hate to create separate component properties for my pivot information, initialize them in `mount()`, and then send them as separate arguments in the event.

**Comment by itsgrimace** [comment]
How about when you pass in your 'Resource' you pass it in using 
`$resource = Resource::with('relationshipName')->find($id);` 
where you include the relationship data and hence the pivot.

**Reply by SturmB** [reply]
Hey, @itsgrimace, thanks for answering.

I'm already eager-loading all relationships in the controller if that's what you mean:

```php
    public function index(): View
    {
        $armorSets = ArmorSet::with([
            "armors.resources" => function ($query) {
                $query->orderBy("tier", "asc");
            },
        ])->get();

        return view("index", compact("armorSets"));
    }
```

And displaying it in the blade template:

```blade
<div>
    @foreach($armorSets as $armorSet)
        <div>
            <h2>{{ $armorSet->name }}</h2>
            @foreach($armorSet->armors as $armor)
                <h3>{{ $armor->name }}</h3>
                @if($armor->upgradable)
                    @foreach($armor->resources->groupBy(fn($resource) => $resource->pivot->tier) as $tierNum => $resources)
                        <p>Tier {{ $tierNum }}:</p>
                        @foreach($resources as $resource)
                            <livewire:resource-checkbox
                                :resource="$resource"
                                :wire:key="'checkbox-' . $resource->pivot->id"
                            />
                        @endforeach
                    @endforeach
                @endif
            @endforeach
        </div>
    @endforeach
</div>
```

**Comment by michael-rubel** [comment]
Hi @SturmB 

I resolved this by reinitializing the object in `booted` method.
This way the pivot is available on subsequent requests.

**Reply by SturmB** [reply]
Hello, @michael-rubel.

How do I go about "reinitializing" the object? It was passed in from a blade template (see my response to itsgrimace, above).

**Reply by michael-rubel** [reply]
@SturmB Try to pass raw data and reinitialize the resource in the component.

P.S.: https://github.com/michael-rubel/livewire-best-practices#-the-golden-rule-of-performant-livewire
